sycl/source/detail/scheduler/leaves_collection.hpp

```diff
@@ -221,6 +221,18 @@ class LeavesCollection {
               (!MGenericIsActive && MHACIt != Rhs.MHACIt));
     }
 
+    IteratorT &operator=(const IteratorT<IsConst> &Rhs) {
```

Comment:
Looks like this copy assignment operator is not used too in the code

Reply:
Coverity reported with messages in below:

2016032 [Copy without assign](https://coverityent.devtools.intel.com/prod6/doc/en/cov_checker_ref.html#static_checker_COPY_WITHOUT_ASSIGN)
This class has a user-defined copy constructor but no user-defined assignment operator. If the copy constructor is necessary to manage owned resources then a corresponding assignment operator is usually required. If an object of this type is assigned memory leaks and/or use-after-free errors may occur. Note that a compiler-generated assignment operator will perform only a bit-wise copy for any fields that do not have their own assignment operators defined.

Class has user-written copy constructor but no user-written assignment operator